cmd/argocd-applicationset-controller/commands/applicationset_controller.go

```diff
@@ -107,6 +108,9 @@ func NewCommand() *cobra.Command {
 			// If the applicationset-namespaces contains only one namespace it corresponds to the current namespace
 			if len(applicationSetNamespaces) == 1 {
 				watchedNamespace = (applicationSetNamespaces)[0]
+			} else if len(allowedScmProviders) == 0 {
+				log.Error("When enabling applicationset in any namespace using applicationset-namespaces, allowed-scm-providers is required")
+				os.Exit(1)
```

Comment:
We should not error out here as we are saying that the empty list of allowedScmProviders allows `all` SCM Providers.

Reply:
Hello :) 
With @crenshaw-dev we agreed on the fact that allowedScmProviders will be required when using applicationset in any namespace has if this is not specified it will allow to exfiltrate secrets

https://github.com/argoproj/argo-cd/pull/14286/files#diff-cec6085c9575a194bca4a245fc032883cb4bca0c720ca5d26f5dbd3acac3463c => `SCM Providers secrets consideration`

Reply:
I also added the following comment in `argocd-cmd-params-cm.yaml`

```yaml 
# Setting this field is required when using ApplicationSets-in-any-namespace, to prevent users from
```